#!/usr/bin/env python
PACKAGE = "thormang3_step_control_module"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

####### Balance enable #####
gen.add("balance",   bool_t,   0, "Balance on/off",  True)

####### cob_offset #######
gen.add("cob_x_offset_m",                   double_t, 0, "", -0.015, -0.1, 0.1)
gen.add("cob_y_offset_m",                   double_t, 0, "",  0,     -0.1, 0.1)

####### FeedForward #####
gen.add("hip_roll_swap_angle_rad",          double_t, 0, "",  0.0,  0,   0.02)

########## Gain ########
# gyro
gen.add("foot_roll_gyro_p_gain",            double_t, 0, "",  0.3,  0,  2)
gen.add("foot_roll_gyro_d_gain",            double_t, 0, "",  0.0,  0,  2)
gen.add("foot_pitch_gyro_p_gain",           double_t, 0, "",  0.3,  0,  2)
gen.add("foot_pitch_gyro_d_gain",           double_t, 0, "",  0.0,  0,  2)

# imu
gen.add("foot_roll_angle_p_gain",           double_t, 0, "",  1.0,  0,  2)
gen.add("foot_roll_angle_d_gain",           double_t, 0, "",  0.0,  0,  2)
gen.add("foot_pitch_angle_p_gain",          double_t, 0, "",  1.0,  0,  2)
gen.add("foot_pitch_angle_d_gain",          double_t, 0, "",  0.0,  0,  2)

# ft
#gen.add("foot_x_force_p_gain",              double_t, 0, "",  0.05,   0,  0.1)
#gen.add("foot_x_force_d_gain",              double_t, 0, "",  0.001,  0,  0.1)
#gen.add("foot_y_force_p_gain",              double_t, 0, "",  0.05,   0,  0.1)
#gen.add("foot_y_force_d_gain",              double_t, 0, "",  0.001,  0,  0.1)
gen.add("foot_x_force_p_gain",              double_t, 0, "",  0.0,   0,  0.1)
gen.add("foot_x_force_d_gain",              double_t, 0, "",  0.0,  0,  0.1)
gen.add("foot_y_force_p_gain",              double_t, 0, "",  0.0,   0,  0.1)
gen.add("foot_y_force_d_gain",              double_t, 0, "",  0.0,  0,  0.1)

gen.add("foot_z_force_p_gain",              double_t, 0, "",  0.02,   0,  0.1)
gen.add("foot_z_force_d_gain",              double_t, 0, "",  0.001,  0,  0.1)

gen.add("foot_roll_torque_p_gain",          double_t, 0, "",  0.0015,  0,  0.1)
gen.add("foot_roll_torque_d_gain",          double_t, 0, "",  0.0002,  0,  0.1)
gen.add("foot_pitch_torque_p_gain",         double_t, 0, "",  0.0015,  0,  0.1)
gen.add("foot_pitch_torque_d_gain",         double_t, 0, "",  0.0002,  0,  0.1)

########## CUT OFF FREQUENCY ##########
# by gyro
gen.add("roll_gyro_cut_off_frequency",          double_t, 0, "",  12.0,  0,  50)
gen.add("pitch_gyro_cut_off_frequency",         double_t, 0, "",  12.0,  0,  50)

# by imu
gen.add("roll_angle_cut_off_frequency",         double_t, 0, "",  12.0,  0,  50)
gen.add("pitch_angle_cut_off_frequency",        double_t, 0, "",  12.0,  0,  50)

# by ft sensor
gen.add("foot_x_force_cut_off_frequency",       double_t, 0, "",  10.0,  0,  40)
gen.add("foot_y_force_cut_off_frequency",       double_t, 0, "",  10.0,  0,  40)
gen.add("foot_z_force_cut_off_frequency",       double_t, 0, "",  10.0,  0,  40)

gen.add("foot_roll_torque_cut_off_frequency",   double_t, 0, "",  10.0,  0,  40)
gen.add("foot_pitch_torque_cut_off_frequency",  double_t, 0, "",  10.0,  0,  40)

exit(gen.generate(PACKAGE, "thormang3", "BalanceParameters"))

